lib/network_util.py

```diff
@@ -6,23 +6,154 @@
 import networkx as nx 
 
 
-class Graph(object):
+class Graph():
+
+    node_dict_factory = dict
+    adjlist_outer_dict_factory = dict
+    adjlist_inner_dict_factory = dict
+    edge_attr_dict_factory = dict
+
+
+    def to_directed_class(self):
+        return nx.DiGraph
+
+    def to_undirected_class(self):
+        return nx.Graph
 
     def __init__(self):
         self.graph = nx.Graph()
+        self.graph.__init__()
+
+    @property
+    def adj(self):
+        return self.graph.adj
+
+    @property
+    def name(self):
+        return self.graph.name
+
+    @name.setter
+    def name(self, s):
+        self.name = s       
+
+    def __str__(self):
+        return self.name
+
+    def __iter__(self):
+        return self.graph.__iter__()
+
+    def __contains__(self, n):
+        return self.graph.__contains__(n)
+
+    def __len__(self):
+        return self.graph.__len__()
+
+    def __getitem__(self, n):
+        self.graph.__getitem__(n)
```

Comment:
having an explicit return statement makes the code more readable and bug free as well.

Reply:
I tried to copy exactly like the networkx graph module source code (https://networkx.github.io/documentation/stable/_modules/networkx/classes/graph.html)